### ISSUE TITLE
getitem shorthand notation consistency

### ISSUE THREAD

**Issue by crowsonkb** [pyparsing/pyparsing#104]
I know it's a bit late for this critique since it's already been released, but in [this commit](https://github.com/pyparsing/pyparsing/commit/4c59256fd3af3206241419b7e8d51abaf9bc8498) a new shorthand notation was introduced for `ZeroOrMore`, `OneOrMore`, and general repetition expressions.

`expr[...] is equivalent to OneOrMore(expr)`

Why not `ZeroOrMore`? You can still get `OneOrMore` via `expr[1, ...]` and it would be consistent with the other getitem shorthand notations:

`expr[n, ...] or expr[n,] is equivalent to expr*n + ZeroOrMore(expr)`

and

`expr[..., n] is equivalent to expr*(0, n)`.

**Comment by crowsonkb** [comment]
And, of course, the old `__mul__` based shorthand behaves thusly:

`expr*(None, None) is equivalent to ZeroOrMore(expr)`.

**Comment by ptmcg** [comment]
I thought about this a bit, and now I'm thinking I jumped the gun with going straight to a 2.4.1 release, without doing an 'a' and maybe a 'b' pre-release. I saw this same inconsistency the morning after pushing to pypi, and was sick about it, as it really does push the indicator that `[...]` should be `ZeroOrMore`, not `OneOrMore`.

I think I will quickly pull 2.4.1, make the changes for this to be ZeroOrMore, and try to address the other issue that was posted about cmd2 (if I can find the source for that). Then put out 2.4.2a for more general exposure and feedback.
